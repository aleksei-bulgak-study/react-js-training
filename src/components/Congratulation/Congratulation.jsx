import React from 'react';
import PropTypes from 'prop-types';
import ModalWindow from '../ModalWindow';

import icon from '../../../public/images/check-circle-regular.svg';
import './styles.css';

const Logo = <img src={icon} className="congratulation__icon" alt="success image logo"/>;

const Congratulation = ({ onClose }) => (
  <div className="congratulation">
    <ModalWindow title="Congratulations!" className="centered" onClose={onClose} Logo={Logo}>
      <p className="congratulation__text">The movie has been added to database successfully</p>
    </ModalWindow>
  </div>
);

Congratulation.propTypes = {
  onClose: PropTypes.func.isRequired,
};

export default Congratulation;
