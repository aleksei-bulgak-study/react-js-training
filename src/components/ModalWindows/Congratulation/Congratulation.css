.centered {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
}

.congratulation__icon {
  width: 5rem;
  margin-bottom: 2rem;
}

.congratulation__text {
  width: 75%;
  margin: 2rem auto;
  text-align: center;
  font-size: 1.3rem;
}
