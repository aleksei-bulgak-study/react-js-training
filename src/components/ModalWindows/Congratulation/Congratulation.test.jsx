import React from 'react';
import { mount } from 'enzyme';
import Congratulation from './Congratulation';

describe('Congratulation', () => {
  test('snapshot', () => {
    const onClose = jest.fn();
    const component = mount(<Congratulation onClose={onClose} />);

    component.find('button.close-button').simulate('click');
    expect(onClose).toHaveBeenCalled();
    expect(component.html()).toMatchSnapshot();
  });
});
