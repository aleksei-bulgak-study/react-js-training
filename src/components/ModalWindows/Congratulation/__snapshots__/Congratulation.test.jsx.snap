// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Congratulation snapshot 1`] = `"<div class=\\"congratulation\\"><div class=\\"modal-window__overlay\\"></div><div class=\\"modal-window\\"><div class=\\"modal-window__wrapper centered\\"><img src=\\"[object Object]\\" class=\\"congratulation__icon\\" alt=\\"success image logo\\"><h1 class=\\"modal-window__title\\">Congratulations!</h1><button type=\\"button\\" class=\\"close-button modal-window__close large\\">Close</button><p class=\\"congratulation__text\\">The movie has been added to database successfully</p></div></div></div>"`;
